languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -36090,19 +36090,146 @@ though, and has since been slightly modified)-->
+                      <token  case_sensitive="yes">ce</token>
+                  </marker>
+                  <token>,</token>
+                  <token postag="[PA].*" postag_regexp="yes"><exception regexp="yes">qu[ie]?</exception></token>
+              </pattern>
+              <message>Une virgule précédant cette expression est conseillée dans ce contexte.</message>
+              <suggestion>\1, \2 \3</suggestion>
+              <url>http://bdl.oqlf.gouv.qc.ca/bdl/gabarit_bdl.asp?id=1505#:~:text=Les%20expressions%20et%20ce%20et%20et%20cela%20s%27emploient,%C3%A9videmment%2C%20qu%E2%80%99ils%20ne%20soient%20en%20t%C3%AAte%20de%20phrase.</url>
+              <example correction="ville, et ce">Elle veut vivre en <marker>ville et ce</marker>, jusqu'à la fin de sa vie.</example>
+              <example>Elle veut vivre en <marker>ville et ce</marker> .</example>
+              <example>Cela donne une indication sur le <marker>terme et ce</marker>, de quoi il est composé.</example>
+          </rule>
+          <rule>
+              <antipattern>
+                  <token>,</token>
+                  <token>et</token>
+                  <token>ce</token>
+                  <token postag="SENT_END"/>
+              </antipattern>
+              <antipattern>
+                  <token>,</token>
+                  <token>et</token>
+                  <token>ce</token>
+                  <token regexp="yes">ne|n(&apostrophe;)|passage|n</token>
+              </antipattern>
+              <antipattern>
+                  <token>et</token>
+                  <token>ce</token>
+                  <token postag="[PA].*" postag_regexp="yes"/>
+                  <token postag="SENT_END"/>
+              </antipattern>
+              <antipattern>
+                  <token>,</token>
+                  <token>et</token>
+                  <token>ce</token>
+                  <token postag="[PA].*" postag_regexp="yes" min="0" max="1"/>
+                  <token regexp="yes">pourquoi|quoi|qu(&apostrophe;)</token>
+              </antipattern>
+              <pattern>
+                  <token regexp="yes">[^,‘’–)\[—."\/;\]\|:(\…'«»“”-]+<exception postag="[PD].*" postag_regexp="yes"/></token>
+                  <marker>
+                      <token>,</token>
+                      <token case_sensitive="yes">et</token>
+                      <token  case_sensitive="yes">ce</token>
+                  </marker>
+                  <token postag="[PA].*" postag_regexp="yes"><exception regexp="yes">qu[ie]?</exception></token>
+              </pattern>
+              <message>Une virgule suivant cette expression est conseillée dans ce contexte.</message>
+              <suggestion>\2 \3 \4,</suggestion>
+              <url>http://bdl.oqlf.gouv.qc.ca/bdl/gabarit_bdl.asp?id=1505#:~:text=Les%20expressions%20et%20ce%20et%20et%20cela%20s%27emploient,%C3%A9videmment%2C%20qu%E2%80%99ils%20ne%20soient%20en%20t%C3%AAte%20de%20phrase.</url>
```

Comment:
I wouldn't use Canadian sources, users might not recognize them as authoritative for European French :)

Reply:
I change the url and extended the scope of the rule.